### PR TITLE
Fabric: add minimal ProgressView component for iOS

### PR DESCRIPTION
**Pull request by ericlewis**
## Summary

Codegen'd ProgressView, this brings RNTester’s example back to life!

## Changelog

[iOS] [Added] - Fabric: ProgressView

## Test Plan

View ProgressViewIOS example in RNTester with Fabric enabled.

### CODE REVIEW

File: React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+#import <UIKit/UIKit.h>
+
+#import <React/RCTViewComponentView.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * UIView class for root <ProgressView> component.
```
Comment (shergin) [inline React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.h]: --root--

File: React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.mm [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  [super updateProps:props oldProps:oldProps];
+
+  // `progress`
+  if (oldProgressViewProps.progress != newProgressViewProps.progress) {
+    _progressView.progress = newProgressViewProps.progress;
+  }
+
+  // `trackTintColor`
+  if (oldProgressViewProps.trackTintColor != newProgressViewProps.trackTintColor) {
+    _progressView.trackTintColor = [UIColor colorWithCGColor:newProgressViewProps.trackTintColor.get()];
```
Comment (shergin) [inline React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.mm]: `.get` is an implementation detail, we have to use special conversion function from RCTConvertions.h`.

File: React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.h (hunk 1 above)
Comment (ericlewis) [reply React/Fabric/Mounting/ComponentViews/ProgressView/RCTProgressViewComponentView.h]: @shergin all the components are like this lol
```
/**
 * UIView class for root <Slider> component.
 */
```

**Comment by ericlewis** [comment]
@shergin I think that addresses your feedback, unless you want me to remove root- but other components have that too, so maybe I could do that in a diff PR..

**Comment by fkgozali** [comment]
In the spirit of Lean Core, I'm not sure we want this in core? So I think we can hold off for now?

**Comment by cpojer** [comment]
Closing based on @fkgozali's comment. Thanks for experimenting with all the new awesome features :)

**Pull request closed without merging**